File: rails_todo_api/app/models/todo.rb
class Todo
    attr_accessor :id, :title, :user_id, :completed, :created_at

    def initialize(id, todo = {})
        self.id = id
        self.title = todo[:title] unless todo.nil?
        self.user_id = todo[:user_id] unless todo.nil?
        self.completed = todo[:completed] unless todo.nil?
        self.created_at = todo[:created_at] unless todo.nil?
    end
end

File: apis/dotnet_todo_api/dotnet_todo_api.core/Models/Todo.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet_todo_api.core.Models
{
    [Table("Todo")]
    public class Todo
    {
        [Key]
        [Required]
        [Column("todo_id", TypeName = "varchar(64)")]
        public required string TodoId { get; set; }

        [Required]
        [Column("title", TypeName = "varchar(256)")]
        public required string Title { get; set; }

        [ForeignKey("User")]
        [Column("user_id", TypeName = "varchar(50)")]
        public required string UserId { get; set; }

        [Required]
        [Column("created_at")]
        public required DateTime CreatedAt { get; set; }

        [Required]
        [Column("completed")]
        public bool Completed { get; set; } = false;

        [NotMapped]
        public virtual required User User { get; set; }
    }
}